src/features/applications/mappers/index.ts

```diff
@@ -109,58 +109,299 @@ const getValue = (bytes: string) => {
+            {
+              prefix: childPrefix,
+            }
+          ),
+      })
+    }
+  }
+  if (type instanceof algosdk.ABIStringType) {
+    return formFieldHelper.textField({
+      label: 'Value',
+      field: `${path}` as Path<TData>,
+      placeholder: options?.description,
+    })
+  }
+  if (type instanceof algosdk.ABITupleType) {
+    const prefix = options?.prefix ?? 'Item'
+    return formFieldHelper.abiTupleField({
+      field: `${path}` as Path<TData>,
+      length: type.childTypes.length,
+      prefix: prefix,
+      description: options?.description,
+      createChildField: (childPrefix, childIndex) =>
+        getCreateField(
+          formFieldHelper,
+          type.childTypes[childIndex],
+          `${path}${arrayItemPathSeparator}${childIndex}` as FieldPath<TData>,
+          undefined,
+          {
+            prefix: childPrefix,
+          }
+        ),
+      struct: hint?.struct,
+    })
+  }
+  if (algosdk.abiTypeIsReference(type)) {
+    return formFieldHelper.numberField({
+      label: options?.prefix ?? 'Value',
+      field: `${path}` as Path<TData>,
+      placeholder: options?.description,
+    })
+  }
+  return undefined
+}
+
+const getAppCallArg = (type: algosdk.ABIArgumentType, value: unknown): ABIAppCallArg => {
+  if (type instanceof algosdk.ABIUintType) {
+    return value as ABIAppCallArg
+  }
+  if (type instanceof algosdk.ABIByteType) {
+    return value as ABIAppCallArg
+  }
+  if (type instanceof algosdk.ABIBoolType) {
+    return value as ABIAppCallArg
+  }
+  if (type instanceof algosdk.ABIUfixedType) {
+    return BigInt((value as number) * Math.pow(10, (type as algosdk.ABIUfixedType).precision)) as ABIAppCallArg
+  }
+  if (type instanceof algosdk.ABIArrayStaticType) {
+    if (type.childType instanceof algosdk.ABIByteType) {
+      return base64ToBytes(value as string) as ABIAppCallArg
+    } else {
+      return (value as unknown[]).map((item) => getAppCallArg(type.childType, item)) as ABIAppCallArg
+    }
+  }
+  if (type instanceof algosdk.ABIAddressType) {
+    return value as ABIAppCallArg
+  }
+  if (type instanceof algosdk.ABIArrayDynamicType) {
+    if (type.childType instanceof algosdk.ABIByteType) {
+      return base64ToBytes(value as string) as ABIAppCallArg
+    } else {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      return (value as any[]).map((item) => getAppCallArg(type.childType, item.child)) as ABIAppCallArg
+    }
+  }
+  if (type instanceof algosdk.ABIStringType) {
+    return value as ABIAppCallArg
+  }
+  if (type instanceof algosdk.ABITupleType) {
+    return (value as unknown[]).map((item, index) => getAppCallArg(type.childTypes[index], item)) as ABIAppCallArg
+  }
+  if (algosdk.abiTypeIsReference(type)) {
+    return value as ABIAppCallArg
+  }
+  return value as ABIAppCallArg
+}
+
+const getDefaultValue = (type: algosdk.ABIArgumentType, isOptional: boolean): unknown => {
+  if (type instanceof algosdk.ABIUintType) {
```

Comment:
```suggestion
  if (type instanceof algosdk.ABIUintType) {
```
Probably want to add support for a the byte type here, as it's a number field too.